sources/core/objects/Grille.cpp

```diff
@@ -37,15 +37,36 @@ void Grille::AttackBy(IdPtr<Item> item)
             SetPassable(D_ALL, Passable::FULL);
             cutted_ = true;
             Create<IOnMapObject>(Rod::T_ITEM_S(), GetOwner());
+            return;
         }
         else
         {
             Create<IOnMapObject>(Rod::T_ITEM_S(), GetOwner());
             Delete();
+            return;
         }
     }
+    Breakable::AttackBy(item);
```

Comment:
Yeah, it definitely the problem. You are right about punch - that's why `AttackByHand` should be reworked #82 (but not right now).
For a while just do not show app the name of the attacker (just sound, or maybe even do not implement it). So disable all features what demand a `Human` info.